<?xml version="1.0" ?>
<!-- Generated using onshape-to-robot -->
<!-- Onshape https://cad.onshape.com/documents/58d69eaf54d5097cb2ee2932/w/94ca09f7b1264f3339bd84c4/e/c562299726437d1b04c86cff -->
<robot name="simplebot2" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:include filename="$(find simplebot_description)/urdf/wheel.xacro"/>
  <!-- Link a010 -->
  <link name="frame4">
    <inertial>
      <origin xyz="-3.43853e-10 -0.0243167 0.0253545" rpy="0 0 0"/>
      <mass value="1.22548"/>
      <inertia ixx="0.00516387" ixy="-9.47574e-11" ixz="-1.56623e-11" iyy="0.00308896" iyz="-0.000170816" izz="0.00729188"/>
    </inertial>
    <!-- Part a010 -->
    <visual>
      <origin xyz="3.46945e-17 -0.085 -0.005" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/a010.stl"/>
      </geometry>
      <material name="a010_material">
        <color rgba="0.231373 0.380392 0.705882 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="3.46945e-17 -0.085 -0.005" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/a010.stl"/>
      </geometry>
    </collision>
    <!-- Part rplidar_c1 -->
    <visual>
      <origin xyz="0 -0.045 -0.005" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/rplidar_c1.stl"/>
      </geometry>
      <material name="rplidar_c1_material">
        <color rgba="0.647059 0.647059 0.647059 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 -0.045 -0.005" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/rplidar_c1.stl"/>
      </geometry>
    </collision>
    <!-- Part frame4 -->
    <visual>
      <origin xyz="7.02021e-18 -0.085 -0.005" rpy="0 -2.31296e-16 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/frame4.stl"/>
      </geometry>
      <material name="frame4_material">
        <color rgba="0.615686 0.811765 0.929412 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="7.02021e-18 -0.085 -0.005" rpy="0 -2.31296e-16 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/frame4.stl"/>
      </geometry>
    </collision>
    <!-- Part top_parts_bracket -->
    <visual>
      <origin xyz="2.08167e-17 -0.045 -0.005" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/top_parts_bracket.stl"/>
      </geometry>
      <material name="top_parts_bracket_material">
        <color rgba="0.917647 0.917647 0.917647 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="2.08167e-17 -0.045 -0.005" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/top_parts_bracket.stl"/>
      </geometry>
    </collision>
    <!-- Part a010_bracket -->
    <visual>
      <origin xyz="3.46945e-17 -0.085 -0.005" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/a010_bracket.stl"/>
      </geometry>
      <material name="a010_bracket_material">
        <color rgba="0.980392 0.713725 0.00392157 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="3.46945e-17 -0.085 -0.005" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="file://$(find simplebot_description)/meshes/test/simplebot/a010_bracket.stl"/>
      </geometry>
    </collision>
  </link>

<xacro:wheel name="wheel_front_left" color="0.1 0.1 0.1 1"/>
  <joint name="joint_front_left" type="continuous">
    <parent link="frame4"/>
    <child link="wheel_front_left"/>
    <origin xyz="0.065 0.103008 0.010" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="5.0" velocity="20.0"/>
  </joint>

  <xacro:wheel name="wheel_front_right" color="0.1 0.1 0.1 1"/>
  <joint name="joint_front_right" type="continuous">
    <parent link="frame4"/>
    <child link="wheel_front_right"/>
    <origin xyz="0.065 -0.103008 0.010" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="5.0" velocity="20.0"/>
  </joint>

  <xacro:wheel name="wheel_rear_left" color="0.1 0.1 0.1 1"/>
  <joint name="joint_rear_left" type="continuous">
    <parent link="frame4"/>
    <child link="wheel_rear_left"/>
    <origin xyz="-0.065 0.103008 0.010" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="5.0" velocity="20.0"/>
  </joint>

  <xacro:wheel name="wheel_rear_right" color="0.1 0.1 0.1 1"/>
  <joint name="joint_rear_right" type="continuous">
    <parent link="frame4"/>
    <child link="wheel_rear_right"/>
    <origin xyz="-0.065 -0.103008 0.010" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="5.0" velocity="20.0"/>
  </joint>

  <gazebo>
    <plugin
    filename="gz-sim-diff-drive-system"
    name="gz::sim::systems::DiffDrive">
      <front_left_joint>joint_front_left</front_left_joint>
      <front_right_joint>joint_front_right</front_right_joint>
      <rear_left_joint>joint_rear_left</rear_left_joint>
      <rear_right_joint>joint_rear_right</rear_right_joint>

      <wheel_separation>0.206</wheel_separation>
      <wheelbase>0.13</wheelbase>
      <wheel_radius>0.03</wheel_radius>

      <topic>/cmd_vel</topic>
      <odom_topic>/odom</odom_topic>
      <frame_id>odom</frame_id>
      <child_frame_id>base_footprint</child_frame_id>

      <min_acceleration>-10</min_acceleration>
      <max_acceleration>10</max_acceleration>
      <min_velocity>-10</min_velocity>
      <max_velocity>10</max_velocity>
    </plugin>
  </gazebo>
</robot>
